src/Quaternion.jl

```diff
@@ -386,3 +386,6 @@ LinearAlgebra.lyap(a::Quaternion, c::Quaternion) = lyap(promote(a, c)...)
 # if a commutes with c, use a simpler expression
 LinearAlgebra.lyap(a::Real, c::Quaternion) = c / -2a
 LinearAlgebra.lyap(a::Quaternion, c::Real) = c / -2real(a)
+
+## RealDot
+@inline RealDot.realdot(p::Quaternion, q::Quaternion) = muladd(p.s, q.s, dot(imag_part(p), imag_part(q)))
```

Comment:
This is almost as verbose as explicitly writing out the sum, so perhaps this is better?
```suggestion
@inline RealDot.realdot(p::Quaternion, q::Quaternion) = p.s * q.s + p.v1 * q.v1 + p.v2 * q.v2 + p.v3 * q.v3
```

Reply:
I was understanding that `muladd` version is more efficient than the suggested explicit version, but the benchmark results are almost the same in my local machine.
Another reason for `muladd` was referring to `realdot(x::Complex, y::Complex)` method in RealDot.jl (https://github.com/JuliaMath/RealDot.jl/blob/6a8e18b23ac2d79d8337b272c78bca7f0ce51bf2/src/RealDot.jl#L17).

Reply:
I used to think `muladd` was faster too, and then IIRC someone corrected me. But now I don't remember. I've never seen a difference in benchmarks. I tend to use `muladd` unless it makes the code much less clear.

@mikmoore do you know anything about this?

Reply:
`muladd` gives the compiler the *option* to use either `a*b+c` or `fma(a,b,c)`. `fma` is more accurate and, in isolation, is faster than the two distinct operations (unless it's not natively supported by the CPU). However, it is sometimes skipped because register shuffling or other things would be required. Note that inlining can change the circumstances and result in different choices as well.